src/Localization/Localization/src/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -153,13 +153,23 @@ public IStringLocalizer Create(Type resourceSource)
                 throw new ArgumentNullException(nameof(resourceSource));
             }
 
-            var typeInfo = resourceSource.GetTypeInfo();
+            // throw new Exception("HERE!");
+            Console.WriteLine("HERE");
 
-            var baseName = GetResourcePrefix(typeInfo);
+            // Get without Add to prevent unnecessary lambda allocation
+            if (_localizerCache.TryGetValue(resourceSource.AssemblyQualifiedName!, out var cache))
+            {
+                return cache;
+            }
 
-            var assembly = typeInfo.Assembly;
+            return _localizerCache.GetOrAdd(resourceSource.AssemblyQualifiedName!, _ =>
```

Comment:
This could just be `Add`, worst case it's called more than once, but only on first concurrent requests.